test-suite/js/src/test/scala/org/scalajs/testsuite/javalib/net/URLDecoderTest.scala

```diff
@@ -0,0 +1,74 @@
+package org.scalajs.testsuite.javalib.math
+
+import java.net.URLDecoder
+import java.io.UnsupportedEncodingException
+import org.scalajs.jasminetest.JasmineTest
+
+object URLDecoderTest extends JasmineTest {
+
+  describe("URLDecoderTest"){
+
+    it("returns the empty string unchanged"){
+      expect(URLDecoder.decode("", "utf-8")).toEqual("")
+    }
+
+    it("transforms + in spaces"){
+      expect(URLDecoder.decode("a+b+c", "utf-8")).toEqual("a b c")
+    }
+
+    it("transforms encoded caracters correctly"){
+      expect(URLDecoder.decode("a%20b%20c", "utf-8")).toEqual("a b c")
+    }
```

Comment:
You should also test sequences of code units longer than 1 byte (such as `%c3%9f`). You can find more of those at https://github.com/scala-js/scala-js/blob/master/test-suite/js/src/test/scala/org/scalajs/testsuite/niocharset/UTF8Test.scala#L23-L38

And also sequences of more than 1 code point (such as `%20%20`)